src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDsa.cs

```diff
@@ -69,9 +69,6 @@ public virtual void GenerateKey(ECCurve curve)
 
         public virtual byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
         {
-            if (data == null)
-                throw new ArgumentNullException(nameof(data));
```

Comment:
Why are you removing these checks?  It's a public method, the checks have to stay.

Reply:
Because the calling method below `return SignData(data, 0, data.Length, hashAlgorithm);` checking exact same condition and throwing exact same exception, but as discussed will revert back

Reply:
In this particular case, the concern is that `SignData` is **virtual**. A user could **override** it and forget to do the check. So, even though the built-in implementation is doing the null check, not all implementations will. Therefore, this check is important to keep.

In other instances, where you have helper/instance methods (that are not virtual), or non-public, you can do the analysis and based on checks higher up in the call graph, remove redundant checks. All **public** entry points should have the check with special attention to **virtual** ones.